stdlib/public/core/Random.swift

```diff
@@ -0,0 +1,152 @@
+/// used to generate the same sequence of random values for testing purposes.
+///
+/// To make a custom type conform to the `RandomNumberGenerator` protocol,
+/// implement the required `next()` method. Each call to `next()` must produce
+/// a uniform and independent random value.
+///
+/// Types that conform to `RandomNumberGenerator` should specifically document
+/// the thread safety and quality of the generator.
+public protocol RandomNumberGenerator {
+  /// Returns a value from a uniform, independent distribution of binary data.
+  ///
+  /// - Returns: An unsigned 64-bit random value.
+  mutating func next() -> UInt64
+}
+
+extension RandomNumberGenerator {
+  /// Returns a value from a uniform, independent distribution of binary data.
+  ///
+  /// - Returns: A random value of `T`. Bits are randomly distributed so that
+  ///   every value of `T` is equally likely to be returned.
+  @inlinable
+  public mutating func next<T: FixedWidthInteger & UnsignedInteger>() -> T {
+    if T.bitWidth <= UInt64.bitWidth {
+      return T(truncatingIfNeeded: self.next())
+    }
+
+    let (quotient, remainder) = T.bitWidth.quotientAndRemainder(
+      dividingBy: UInt64.bitWidth
+    )
+    var tmp: T = 0
+
+    for i in 0 ..< quotient {
+      tmp += T(truncatingIfNeeded: self.next()) &<< (UInt64.bitWidth * i)
+    }
+
+    if remainder != 0 {
+      let random = self.next()
+      let mask = UInt64.max &>> (UInt64.bitWidth - remainder)
+      tmp += T(truncatingIfNeeded: random & mask) &<< (UInt64.bitWidth * quotient)
+    }
+
+    return tmp
+  }
+
+  /// Returns a random value that is less than the given upper bound.
+  ///
+  /// - Parameter upperBound: The upper bound for the randomly generated value.
+  /// - Returns: A random value of `T` in the range `0..<upperBound`. Every
+  ///   value in the range `0..<upperBound` is equally likely to be returned.
+  @inlinable
+  public mutating func next<T: FixedWidthInteger & UnsignedInteger>(
+    upperBound: T
+  ) -> T {
+    let tmp = (T.max % upperBound) + 1
+    let range = tmp == upperBound ? 0 : tmp
+    var random: T = 0
+
+    repeat {
+      random = self.next()
+    } while random < range
+
+    return random % upperBound
+  }
+}
+
+/// The default source of random data.
+///
+/// When you generate random values, shuffle a collection, or perform another
+/// operation that depends on random data, this type's `default` property is
+/// the generator used by default. For example, the two method calls in this
+/// example are equivalent:
+///
+///     let x = Int.random(in: 1...100)
+///     let y = Int.random(in: 1...100, using: &Random.default)
+///
+/// `Random.default` is safe to use in multiple threads, and uses a
+/// cryptographically secure algorithm whenever possible.
+public struct Random : RandomNumberGenerator {
+  /// The shared, default instance of the `Range` random number generator.
+  public static var `default` = Random()
```

Comment:
@jckarter Does this need to be the implementation you gave here? https://forums.swift.org/t/se-0202-random-unification/11313/171

Reply:
The core team decided that passing an explicit default is likely rare enough not to need the no-op setter to begin with.